content/spire/try/oidc-federation-aws.md

```diff
@@ -0,0 +1,415 @@
+           "Federated": "arn:aws:iam::012345678901:oidc-provider/MY_DISCOVERY_DOMAIN"
+         },
+         "Action": "sts:AssumeRoleWithWebIdentity",
+         "Condition": {
+           "StringEquals": {
+             "MY_DISCOVERY_DOMAIN:aud": "mys3",
+             "MY_DISCOVERY_DOMAIN:sub": "spiffe://example.org/ns/default/sa/default"
+           }
+         }
+       }
+     ]
+   }
+   ```
+
+5. Click **Update Trust Policy**. This change to the IAM role takes a minute or two to propagate.
+
+## Create an AWS S3 Test File
+
+Create a simple test file in an AWS S3 bucket to use for testing.
+
+1. Create a text file on your local computer called `test.txt` with the following single line:
+
+   ```console
+   oidc-tutorial file
+   ```
+
+   You'll upload this file to the S3 bucket momentarily.
+
+2. Navigate to the AWS [Amazon S3 page](https://s3.console.aws.amazon.com/s3/home), logging in if necessary.
+
+3. Click **Create bucket**.
+
+4. Under **Bucket name** type `oidc-federation-test-bucket`.
+
+5. Leave the **Region**, **Bucket settings for Block Public Access**, and **Advanced settings** at the default values and click **Create bucket**.
+
+6. Click `oidc-federation-test-bucket` to open the bucket. 
+
+7. Click **Upload**.
+
+8. Add the `test.txt` file to the upload area using your local file navigator or drag and drop.
+
+9. Click **Upload**.
+
+# Part Three: Test Access to AWS S3
+
+Now that Kubernetes and AWS are configured for OIDC federation, we'll test the connection. This test retrieves a JWT SVID from the SPIRE Agent and uses the JWT token in the JWT SVID to access S3.
+
+1. Run the following command to list the full name of the client pod:
+   ```console
+   $ kubectl get pods
+
+   NAME                      READY   STATUS    RESTARTS   AGE
+   client-74d4467b44-7nrs2   1/1     Running   0          27s
+   ```
+
+2. Start a shell on the client using the client pod name shown under **NAME** in the output of the last step.
+
+   ```console
+   $ kubectl exec -it client-74d4467b44-7nrs2 /bin/sh
+   ```
+
+3. Fetch a JWT SVID from the identity provider on AWS and save the token from the JWT SVID into a file on the client container called `token`:
+
+   ```console
+   # /opt/spire/bin/spire-agent api fetch jwt -audience mys3 -socketPath /run/spire/sockets/agent.sock | sed '2!d' | sed 's/[[:space:]]//g' > token
+   ```
+
+4. Navigate to the AWS [Identity and Access Management (IAM) page](https://console.aws.amazon.com/iam/home#/home), logging in if necessary.
+
+5. Click **Roles**, search for the `oidc-federation-test-role` role you created, and click on the role name.
+
+6. At the top of the **Summary** page, next to **Role ARN**, copy the role ARN into the clipboard by clicking the small icon at the end.
+
+7. Run the following command, pasting the ARN in place of ROLE-NAME-ARN:
+
+   ```console
+   # AWS_ROLE_ARN=ROLE-NAME-ARN AWS_WEB_IDENTITY_TOKEN_FILE=token aws s3 cp s3://oidc-federation-test-bucket/test.txt test.txt
```

Comment:
Consider making these two lines for readability. ie.
```
# AWS_ROLE_ARN=ROLE-NAME-ARN 
# AWS_WEB_IDENTITY_TOKEN_FILE=token aws s3 cp s3://oidc-federation-test-bucket/test.txt
```

Reply:
Also I couldn't complete the exercise any further because I got the error "fatal error: Unable to locate credentials"

```
# echo $AWS_ROLE_ARN
arn:aws:iam::867136367285:role/oidc-federation-test-role
# echo $AWS_WEB_IDENTITY_TOKEN_FILE
token
cat $AWS_WEB_IDENTITY_TOKEN_FILE
eyJhbGciOiJFUzI1NiIsImtpZCI6Ik1PZ0NxU0loMElYcDlpWlBPbjNwWEhzQU5GcDlaY1E2IiwidHlwIjoiSldUIn0.eyJhdWQiOlsibXlzMyJdLCJleHAiOjE1ODU0Mjc4MjAsImlhdCI6MTU4NTQyNzUyMCwiaXNzIjoiaHR0cHM6Ly9vaWRjLmVsZWN0cmljLmRpcmVjdG9yeSIsInN1YiI6InNwaWZmZTovL2V4YW1wbGUub3JnL25zL2RlZmF1bHQvc2EvZGVmYXVsdCJ9.KUWZqeFinkfzVOImN7NeZ5SekPHh4281aEX5vQLZpQMoYaSRmTgOvzob_OZzbBb3Jm1yFvgslgYXXdKfmQl-Bw
```

Reply:
I'm sorry it's not working for you. I did test it myself and got it working, though not after a snafu with a mismatched bucket name that was my error. 
The command seems to work with the environment variables on the same line as the command. Maybe it's easier for users to copy/paste one line from the doc than three?

Reply:
I got this working once I created the bucket, and ran everything on the same line rather than creating the environment variables on separate lines.